lib/library.rb

```diff
@@ -0,0 +1,68 @@
+class Library
+  attr_reader :name,
+              :books,
+              :authors,
+              :checked_out_books,
+              :times_checked_out
+  def initialize(name)
+    @name = name
+    @books = []
+    @authors = []
+    @checked_out_books = []
+    @times_checked_out = {}
+  end
+
+  def add_author(author)
+    @authors << author
+    books << author.books
+    books.flatten!
```

Comment:
This works, but what if the author writes a new book after they're added to the library? How would we add that in? If we calculate the books based off of the @authors instance variable, we'll be sure to always get all the books.